clients/da-vinci-client/src/main/java/com/linkedin/davinci/blob/BlobSnapshotManager.java

```diff
@@ -0,0 +1,139 @@
+package com.linkedin.davinci.blob;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.store.rocksdb.RocksDBUtils;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import com.linkedin.venice.utils.locks.AutoCloseableLock;
+import java.io.File;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.rocksdb.Checkpoint;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+
+
+public class BlobSnapshotManager {
+  private final Map<String, Map<Integer, AtomicLong>> concurrentSnapshotUsers;
+  private Map<String, Map<Integer, Long>> snapShotTimestamps;
+  private final long snapshotRetentionTime;
+  private final String basePath;
+  private final Lock lock = new ReentrantLock();
+  private static final Logger LOGGER = LogManager.getLogger(BlobSnapshotManager.class);
+
+  /**
+   * Constructor for the BlobSnapshotManager
+   */
+  public BlobSnapshotManager(String basePath, long snapshotRetentionTime) {
+    this.basePath = basePath;
+    this.snapshotRetentionTime = snapshotRetentionTime;
+    this.snapShotTimestamps = new VeniceConcurrentHashMap<>();
+    this.concurrentSnapshotUsers = new VeniceConcurrentHashMap<>();
+  }
+
+  public void setSnapShotTimestamps(Map<String, Map<Integer, Long>> snapShotTimestamps) {
```

Comment:
Should this be a public method?
I think this should only be used in test.
If so, could you remove the `public` modifier, and move it down?

Reply:
will do that